### ISSUE TITLE
Single element string array is considered as field selector

### ISSUE THREAD

**Issue by masaruh** [elastic/logstash#3733]
Conditional like this
```
if [status] in ["starts"] {
        :
```

is compiled as:
```
 if (((x = event["[\"starts\"]"]; x.respond_to?(:include?) && x.include?(event["[status]"])))) # if [status] in ["starts"]
```
instead of:
```
 if (((x = [("starts")]; x.respond_to?(:include?) && x.include?(event["[status]"]))))
```

I believe it's because single element array is recognized as field selector.

It looks excluding `"` in `selector_element` (https://github.com/elastic/logstash/blob/master/lib/logstash/config/grammar.treetop#L236) as well fixes this particular case.
(I assume it won't help in the case of single numeric element array, though)

Is it feasible?

**Comment by jordansissel** [comment]
Indeed this is a bug, and there's an open PR proposing a fix (https://github.com/elastic/logstash/pull/1361) but it hasn't been reviewed yet.

**Comment by jordansissel** [comment]
I'd like to see it fixed, but I think my hesitation on #1361 was mostly that I haven't yet had energy to consider what, if any, negative impacts will happen to folks existing configs if they are relying on this accidental behavior, and how we could help surface these problems to users while also fixing it.

**Comment by jsvd** [comment]
closing as dup of https://github.com/elastic/logstash/issues/1850
